### PR TITLE
Update octokit/rest@16.34.1

### PR DESCRIPTION
**Pull request by cristianooliveira**
## Description

Bump `octokit/rest` in order to remove a high severity vulnerability from an old package.
Updated code and tests accordingly.

Vulnerability info can be seen here: https://www.npmjs.com/advisories/1184

```
Before:
2 vulnerabilities found - Packages audited: 874956
Severity: 1 Moderate | 1 High

After:
1 vulnerabilities found - Packages audited: 875046
Severity: 1 Moderate
```

### CODE REVIEW

File: src/changelog-fetcher.js [hunk 1]
```diff
@@ -4,16 +4,10 @@
  * Module dependencies.
  */
 
-const { assign, chain, flatMap, find, isEmpty, range } = require('lodash');
+const _ = require('lodash');
```
Comment (waldyrious) [inline src/changelog-fetcher.js]: Please split the lodash importing changes into a separate commit, to avoid polluting the diff of the actual functional changes due to the upgrade.

File: src/changelog-fetcher.js [hunk 2]
```diff
@@ -70,17 +70,17 @@ class ChangelogFetcher {
 
     const prs = await this.client.paginate(options);
 
-    return chain(prs)
+    return _.chain(prs)
```
Comment (rplopes) [inline src/changelog-fetcher.js]:
```suggestion
    return _(prs)
```

On other uses of `_.chain()` as well.

File: src/changelog-fetcher.js [hunk 3]
```diff
@@ -70,17 +70,17 @@ class ChangelogFetcher {
...
-          !chain(labels)
+          _.isEmpty(this.labels) ||
+          !_.chain(labels)
             .map('name')
             .intersection(this.labels)
             .isEmpty()
             .value()
       )
-      .map(pr => assign(pr, { merged_at: moment.utc(pr.merged_at) }))
+      .map(pr => _.assign(pr, { merged_at: moment.utc(pr.merged_at) }))
```
Comment (rplopes) [inline src/changelog-fetcher.js]: Since this project is now requiring node >= v8, WDYT about replacing  `_.assign(foo, { bar: 'biz' })` with `{ ...foo, bar: 'biz' }`?

File: src/changelog-fetcher.js [hunk 4]
```diff
@@ -35,7 +35,7 @@ class ChangelogFetcher {
    */
 
   assignPullRequestToRelease(releases, pr) {
-    const release = find(releases, release => pr.merged_at.isSameOrBefore(release.created_at));
+    const release = _.find(releases, release => pr.merged_at.isSameOrBefore(release.created_at));
```
Comment (rplopes) [inline src/changelog-fetcher.js]: Maybe we can replace `_.find()` with the native alternative?

```suggestion
    const release = releases.find(release => pr.merged_at.isSameOrBefore(release.created_at));
```

File: src/changelog-fetcher.js [hunk 5]
```diff
@@ -65,23 +59,18 @@ class ChangelogFetcher {
    */
 
   async getAllPullRequests() {
-    const options = {
+    const options = this.client.pulls.list.endpoint.merge({
```
Comment (rplopes) [inline src/changelog-fetcher.js]: If `options` is now only being used in one place, maybe we can inline it there?

Same for the one in releases.

File: src/changelog-fetcher.js [hunk 6]
```diff
@@ -65,33 +59,28 @@ class ChangelogFetcher {
...
       base: this.base,
       owner: this.owner,
       page: 1,
       per_page: 100,
       repo: this.repo,
       state: 'closed'
-    };
-    const prs = await this.client.pullRequests.list(options);
-    const nextPages = await Promise.map(
```
Comment (rplopes) [inline src/changelog-fetcher.js]: Since this module is no longer using `Promise.map()`, it should be safe to stop requiring `bluebird` 🎉.

File: src/changelog-fetcher.js (hunk 4 above)
Comment (cristianooliveira) [reply src/changelog-fetcher.js]: Sure

File: src/changelog-fetcher.js (hunk 6 above)
Comment (cristianooliveira) [reply src/changelog-fetcher.js]: Will remove it

File: src/changelog-fetcher.js [hunk 7]
```diff
@@ -59,28 +58,28 @@ class ChangelogFetcher {
...
+      })
+    );
+
+    return _(prs)
       .filter(
         ({ labels }) =>
-          isEmpty(this.labels) ||
-          !chain(labels)
+          _.isEmpty(this.labels) ||
+          !_.chain(labels)
```
Comment (rplopes) [inline src/changelog-fetcher.js]: Missing replacing `_.chain()` with `_()` here and in 109 as well.

File: src/changelog-fetcher.js [hunk 8]
```diff
@@ -107,8 +106,8 @@ class ChangelogFetcher {
       });
     }
 
-    return chain(releases)
-      .map(release => assign(release, { created_at: moment.utc(release.created_at), prs: [] }))
+    return _.chain(releases)
+      .map(release => _.assign(release, { created_at: moment.utc(release.created_at), prs: [] }))
```
Comment (rplopes) [inline src/changelog-fetcher.js]: Missing replacing `_.assign()` with native object destructuring here as well.

File: yarn.lock [hunk 9]
```diff
@@ -722,10 +722,6 @@ before-after-hook@^2.0.0:
   resolved "https://registry.yarnpkg.com/before-after-hook/-/before-after-hook-2.1.0.tgz#b6c03487f44e24200dd30ca5e6a1979c5d2fb635"
   integrity sha512-IWIbu7pMqyw3EAJHzzHbWa85b6oud/yfKYg5rqB5hNE8CeMi3nX+2C2sj0HswfblST86hpVEOAb9x34NZd6P7A==
 
-bluebird@^3.5.3:
```
Comment (rplopes) [inline yarn.lock]: Maybe do this in a separate commit, since it's not related to lodash?
